### PR TITLE
fix: validation error check for ts code generation

### PR DESCRIPTION
**Pull request by zcstarr**

### CODE REVIEW

File: templates/typescript/templated/exported-class.template.ts [hunk 1]
```diff
@@ -87,7 +87,8 @@ export class <%= className %> {
   private request(methodName: string, params: any[]): Promise<any> {
     const methodObject = _.find(this.openrpcDocument.methods, ({name}) => name === methodName) as MethodObject;
     const openRpcMethodValidationErrors = this.validator.validate(methodName, params);
-    if (openRpcMethodValidationErrors.length > 0) {
+    const isErrorArray = Array.isArray(openRpcMethodValidationErrors) && openRpcMethodValidationErrors.length > 0;
```
Comment (BelfordZ) [inline templates/typescript/templated/exported-class.template.ts]: or alternatively, 
```
if (
  openRpcMethodValidationErrors instanceof MethodNotFoundError || 
  openRpcMethodValidationErrors.length > 0
) {
  ...
}
```

**Comment by BelfordZ** [comment]
while I think the instanceof check is more succinct, feel free to merge away.

**Comment by zcstarr** [comment]
> while I think the instanceof check is more succinct, feel free to merge away.

Yeah I like the brevity of instanceof, but I know I have an irrational fear of it :). It's cleaner so I've done 👍

File: templates/typescript/templated/exported-class.template.ts [hunk 2]
```diff
@@ -85,10 +84,10 @@ export class <%= className %> {
   }
 
   private request(methodName: string, params: any[]): Promise<any> {
-    const methodObject = _.find(this.openrpcDocument.methods, ({name}) => name === methodName) as MethodObject;
+    const methodObject = _.find(<%= className %>.openrpcDocument.methods, ({name}) => name === methodName) as MethodObject;
     const openRpcMethodValidationErrors = this.validator.validate(methodName, params);
-    if (openRpcMethodValidationErrors.length > 0) {
-      return Promise.reject(openRpcMethodValidationErrors);
+    if ( openRpcMethodValidationErrors instanceof MethodNotFoundError || openRpcMethodValidationErrors.length > 0) {
+        return Promise.reject(openRpcMethodValidationErrors);
```
Comment (shanejonas) [inline templates/typescript/templated/exported-class.template.ts]:
```suggestion
      return Promise.reject(openRpcMethodValidationErrors);
```

**Pull request merged**